File: CPP/PROGRAM/Pelanggan.cpp
// KELAS PELANGGAN
#pragma once
#include <iostream>
#include <string>
#include "Manusia.cpp"
#include "Laptop.cpp"

using namespace std;

class Pelanggan : public Manusia {
    private:
        string idPelanggan;
        Laptop laptop;
    public:
        Pelanggan() {}
        
        Pelanggan(string nama, string noTelepon, string alamat, string idPelanggan) 
            : Manusia(nama, noTelepon, alamat) {
            this->idPelanggan = idPelanggan;
        }
        
        void setIdPelanggan(string idPelanggan) {
            this->idPelanggan = idPelanggan;
        }
        
        void setLaptop(Laptop laptop) {
            this->laptop = laptop;
        }
        
        string getIdPelanggan() {
            return idPelanggan;
        }
        
        Laptop getLaptop() {
            return laptop;
        }
        
        void tampilkanInfo() {
            cout << "ID Pelanggan: " << idPelanggan << endl;
            Manusia::tampilkanInfo();
            
            if (laptop.getNama().size() > 0) {
                cout << "\n[ SPESIFIKASI LAPTOP PELANGGAN ]" << endl;
                laptop.tampilkanInfo();
            }
        }
    
    ~Pelanggan() {}
};

File: PYTHON/PROGRAM/Pelanggan.py
# KELAS PELANGGAN
from Manusia import Manusia
from Laptop import Laptop

class Pelanggan(Manusia):
    def __init__(self, nama="", no_telepon="", alamat="", id_pelanggan=""):
        Manusia.__init__(self, nama, no_telepon, alamat)
        self.__id_pelanggan = id_pelanggan
        self.__laptop = Laptop()
    
    def set_id_pelanggan(self, id_pelanggan):
        self.__id_pelanggan = id_pelanggan
    
    def set_laptop(self, laptop):
        self.__laptop = laptop
    
    def get_id_pelanggan(self):
        return self.__id_pelanggan
    
    def get_laptop(self):
        return self.__laptop
    
    def tampilkan_info(self):
        print("ID Pelanggan:", self.__id_pelanggan)
        Manusia.tampilkan_info(self)
        
        if len(self.__laptop.get_nama()) > 0:
            print("\n[ SPESIFIKASI LAPTOP PELANGGAN ]")
            self.__laptop.tampilkan_info()